src/target/parser.rs

```diff
@@ -0,0 +1,130 @@
+use std::{env, str::FromStr};
+
+use crate::{
+    target::{llvm, TargetInfo},
+    utilities::OnceLock,
+    Error, ErrorKind,
+};
+
+#[derive(Debug)]
+struct TargetInfoParserInner {
```

Comment:
Nit:
```suggestion
struct ParsedTargetInfo {
```

(Or `OwnedTargetInfo`)

Reply:
Hmmmm I think `Inner` suffix is ok, if we want to rename it we probably want to rename the other one as well.